### PR TITLE
Added the proper cursor usage flags

### PR DESCRIPTION
**Pull request by pallavigopal**
The cursor usage flag is set to BO_USE_NONE instead of BO_USE_CURSOR.
If i try to change the usage properly to BO_USE_CURSOR the cursor allocations
are failing.

Fixed the allocation failure by adding the BO_USE_CURSOR usage flags with the
DRM_FORMAT_ABGR8888 in the i915 supported combinations.

Jira: AIA-163
Tests: Boots to home screen with mouse connected without any allocation failures
Signed-off-by: Pallavi G <pallavi.g@intel.com>

### CODE REVIEW

File: i915.c [hunk 1]
```diff
@@ -21,7 +21,7 @@ static struct supported_combination combos[19] = {
 	{DRM_FORMAT_ARGB1555, DRM_FORMAT_MOD_NONE,
 		BO_USE_RENDERING | BO_USE_SW_READ_RARELY | BO_USE_SW_WRITE_RARELY},
 	{DRM_FORMAT_ABGR8888, DRM_FORMAT_MOD_NONE,
```
Comment (kalyankondapally) [inline i915.c]: Could this be related https://github.com/android-ia/minigbm/blob/master/cros_gralloc/cros_gralloc_module.cc#L276 ?

Comment (kalyankondapally) [inline i915.c]: Also, I see we are removing BO_USE_RENDERING this here. I am wondering if we didnt hardcode the format in  https://github.com/android-ia/minigbm/blob/master/cros_gralloc/cros_gralloc_module.cc#L276 do we still need this.

Comment (pallavigopal) [inline i915.c]: yes related to this format forcing but not sure why we need this. I can add the BO_USE_RENDERING also part of the usage.

Comment (kalyankondapally) [inline i915.c]: Hmm, Can you check what happens if we remove the format forcing which we have now ?

Comment (pallavigopal) [inline i915.c]: Am confused with this ARGB/ABGR. with/without hardcode we are getting the cursor buffer format as DRM_FORMAT_ABGR8888. In the supported combinations the BO_USE_CURSOR is not part of the usage flags for the DRM_FORMAT_ABGR8888.

Comment (pallavigopal) [inline i915.c]: one more doubt i have when this cros_gralloc_perform get called after the allocation ?
If its after the allocation then its the expected behaviour

Comment (kalyankondapally) [inline i915.c]: ya chromeos doesnt use ABGR format for Cursor, so might have not had it. cros_gralloc_perform gets called after allocation. I think if you can bring back BO_USE_RENDERING and prefix AndroidIA in commit message indicating we need to maintain this patch.

**Comment by pallavigopal** [comment]
Need to rebase with suggested changes.

**Pull request closed without merging**